lightning/src/routing/router.rs

```diff
@@ -200,11 +342,20 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
 	let mut targets = BinaryHeap::new(); //TODO: Do we care about switching to eg Fibbonaci heap?
 	let mut dist = HashMap::with_capacity(network.get_nodes().len());
 
+	// When arranging a route, we select multiple paths so that we can make a multi-path payment.
+	// Don't stop searching for paths when we think they're sufficient to transfer a given value aggregately.
+	// Search for higher value, so that we collect many more paths, and then select the best combination among them.
+	const ROUTE_CAPACITY_PROVISION_FACTOR: u64 = 4;
+	let recommended_value_msat = final_value_msat * ROUTE_CAPACITY_PROVISION_FACTOR as u64;
+
+	// Step (1).
+	// Prepare the data we'll use for payee-to-payer search by inserting first hops suggested by the caller as targets.
+	// Our search will then attempt to reach them while traversing from the payee node.
 	let mut first_hop_targets = HashMap::with_capacity(if first_hops.is_some() { first_hops.as_ref().unwrap().len() } else { 0 });
 	if let Some(hops) = first_hops {
 		for chan in hops {
 			let short_channel_id = chan.short_channel_id.expect("first_hops should be filled in with usable channels, not pending ones");
-			if chan.remote_network_id == *target {
+			if chan.remote_network_id == *payee {
 				return Ok(Route {
```

Comment:
Might it be the case that there's not enough bandwidth in our direct channel, and we need additional routes to get the full payment to `payee`?

Reply:
This is same at it was before this PR, so the expectation is it's the caller's responsibility to check first_hops.
@TheBlueMatt ?

Reply:
But we consider it much latter, in add_entry!, which is not even called here, we just return right away.